pkg/test/framework/components/authz/server.go

```diff
@@ -28,6 +35,27 @@ type Server interface {
 	Providers() []Provider
 }
 
+type deploymentMergeSpec struct {
```

Comment:
why not just use an actual k8s spec? You can just parse the yaml into an actual k8s `Deployment`, make the changes, and then re-marshal to yaml. No merge-patch necessary.